### DISCUSSION TITLE
Deploying a zip file

### DISCUSSION THREAD

**Discussion by MarcL** [vercel/vercel#4426] [Help] [answered]
Hi all,

Firstly, thanks for a great platform. I've used Zeit/Vercel for a while now and I love using it.

I know this is a developer-focussed platform but I'm wondering if there's a way to easily deploy a zip file containing code for some serverless functions. I'd like to give away (or maybe sell) a project I've written but to non-technical people. It would be available as a zip file on something like [Gumroad](https://gumroad.com/). Ideally, I'd like to be able to have simple instructions to allow them to deploy the application to their own Vercel account but wanted to avoid the need to install any developer tools (Node, `now` etc.)

I could use a deployment button but I'd want to keep the repository private if I was selling the project.

Any ideas for a good way to create a new per-user deployment from my code?

Thanks,
Marc

**Comment by styfle** [comment, marked as answer]
You could create a private repo and when someone purchases a license, you could invite them to the repo and they could click the fork button and they would have their own private repo. Then they could sync with the upstream if you wanted to provide changes to the code.

The only thing they would need to do to deploy is create a Vercel account and connect the Git Integration.

**Reply by MarcL** [reply]
That's an interesting plan @styfle - thanks!
I'm just trying to make it as easy as possible for them to deploy so that could work. :)
